### PR TITLE
drivers: i2c: DW i2c: use 32 bit access instead of 16 and 32 bit mix

### PR DESCRIPTION
**Pull request by IRISZZW**
Current DW I2C driver uses 32 bit access for some registers and
16 bit access for others. So if DW I2C IP is connected via bus
which doesn't support 16 bit access we will get bus error.

Fix that by switching to 32 bit access only instead of 16 and 32 bit mix.

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

File: drivers/i2c/i2c_dw_registers.h [hunk 1]
```diff
@@ -13,62 +13,27 @@ extern "C" {
...
-		uint16_t         master_mode : 1 __packed;
-		uint16_t         speed : 2 __packed;
-		uint16_t         addr_slave_10bit : 1 __packed;
-		uint16_t         addr_master_10bit : 1 __packed;
-		uint16_t         restart_en : 1 __packed;
-		uint16_t         slave_disable : 1 __packed;
-		uint16_t         stop_det : 1 __packed;
-		uint16_t         tx_empty_ctl : 1 __packed;
-		uint16_t         rx_fifo_full : 1 __packed;
+		uint32_t         master_mode : 1 __packed;
```
Comment (pabigot) [inline drivers/i2c/i2c_dw_registers.h:19]: Well, here's an underlying problem: It is implementation-defined (C17 J.3.9) in C whether a bit-field type is anything other than a `signed int`, `unsigned int`, or `_Bool` (C17 6.7.2.1).  So this should probably be `unsigned int`.

Hopefully the endianness of the target platforms is consistent with the bit ordering used by the toolchain.

**Review by pabigot — APPROVED** [review]
Seems reasonable, though I haven't checked in detail.  And this use of bit-fields to break out hardware register fields is not necessarily robust across toolchains.

File: drivers/i2c/i2c_dw_registers.h (hunk 1 above)
Comment (ruuddw) [reply drivers/i2c/i2c_dw_registers.h:19]: With the change of driver style from bitfield struct to read/write access functions, do these remaining fields/structs need to be bitfields/packed at all?

**Comment by ruuddw** [comment]
This type of driver style with read/write access functions is more robust, and also consistent with other DesignWare drivers like SPI.
Just wonder why there still are some bitfields/packed structs?

**Pull request merged**